colossalai/nn/parallel/data_parallel.py

```diff
@@ -283,7 +283,9 @@ def _setup_grads_ptr(self):
             p.grad = None
 
     def _post_backward(self):
-        assert self.chunk_manager.accessed_mem == 0
+        if self.chunk_manager.accessed_mem != 0:
```

Comment:
Can you add assert in grad hook to ensure tensor's state == HOLD_AFTER_BWD?

Reply:
done.